hoomd/md/TwoStepRATTLEBD.h

```diff
@@ -0,0 +1,500 @@
+                    Scalar tolerance = 0.000001
+                    );
+
+        virtual ~TwoStepRATTLEBD()
+        {
+        m_exec_conf->msg->notice(5) << "Destroying TwoStepRATTLEBD" << endl;
+        }
+
+        //! Performs the first step of the integration
+        virtual void integrateStepOne(uint64_t timestep);
+
+        //! Performs the second step of the integration
+        virtual void integrateStepTwo(uint64_t timestep);
+
+        //! Includes the RATTLE forces to the virial/net force
+        virtual void includeRATTLEForce(uint64_t timestep);
+
+        //! Get the number of degrees of freedom granted to a given group
+        virtual Scalar getTranslationalDOF(std::shared_ptr<ParticleGroup> group)
+        {
+        // get the size of the intersection between query_group and m_group
+        unsigned int intersect_size = ParticleGroup::groupIntersection(group, m_group)->getNumMembersGlobal();
+
+        return Manifold::dimension() * intersect_size;
+        }
+
+        /// Sets tolerance
+        void setTolerance(Scalar tolerance){ m_tolerance = tolerance; };
+
+        /// Gets tolerance
+        Scalar getTolerance(){ return m_tolerance; };
+
+    protected:
+
+        Manifold m_manifold;  //!< The manifold used for the RATTLE constraint
+        bool m_noiseless_t;
+        bool m_noiseless_r;
+        Scalar m_tolerance;                      //!< The tolerance value of the RATTLE algorithm, setting the tolerance to the manifold
+    };
+
+
+
+/*! \file TwoStepRATTLEBD.h
+    \brief Contains code for the TwoStepRATTLEBD class
+*/
+
+/*! \param sysdef SystemDefinition this method will act on. Must not be NULL.
+    \param group The group of particles this integration method is to work on
+    \param manifold The manifold describing the constraint during the RATTLE integration method
+    \param T Temperature set point as a function of time
+    \param use_alpha If true, gamma=alpha*diameter, otherwise use a per-type gamma via setGamma()
+    \param alpha Scale factor to convert diameter to gamma
+    \param noiseless_t If set true, there will be no translational noise (random force)
+    \param noiseless_r If set true, there will be no rotational noise (random torque)
+    \param tolerance Tolerance for the RATTLE iteration algorithm
+*/
+template< class Manifold>
+TwoStepRATTLEBD<Manifold>::TwoStepRATTLEBD(std::shared_ptr<SystemDefinition> sysdef,
+                           std::shared_ptr<ParticleGroup> group,
+                           Manifold manifold,
+                           std::shared_ptr<Variant> T,
+                           Scalar tolerance
+                           )
+  : TwoStepLangevinBase(sysdef, group, T), m_manifold(manifold),
+    m_noiseless_t(false), m_noiseless_r(false), m_tolerance(tolerance)
+    {
+    m_exec_conf->msg->notice(5) << "Constructing TwoStepRATTLEBD" << endl;
+
+    bool manifold_fits = m_manifold.fitsInsideBox(m_pdata->getBox());
```

Comment:
Shouldn't this be using `m_pdata->getGlobalBox()`? @SchoeniPhlippsn

Reply:
Yes!